### ISSUE TITLE
Getting 500 INTERNAL_SERVER_ERROR  on SSR page

### ISSUE THREAD

**Issue by lukebennett88** [gladly-team/next-firebase-auth#114]
**Describe the bug**

I've had a user let me know that they are getting a 500 Internal Server error when trying to hit an SSR page that uses `next-firebase-auth`.

Not sure if this is me doing something wrong or a bug as I can't seem to replicate this on my end.

I checked the logs in Vercel and saw this:

```shell
[GET] /cart
15:29:43:52
2021-03-16T04:29:45.461Z	a5539b48-1846-4b8a-8914-d28700cb65a6	ERROR	Unhandled error during request: FirebaseAuthError: Firebase ID token has "kid" claim which does not correspond to a known public key. Most likely the ID token is expired, so get a fresh token from your client app and try again.
    at FirebaseAuthError.FirebaseError [as constructor] (/var/task/node_modules/firebase-admin/lib/utils/error.js:44:28)
    at FirebaseAuthError.PrefixedFirebaseError [as constructor] (/var/task/node_modules/firebase-admin/lib/utils/error.js:90:28)
    at new FirebaseAuthError (/var/task/node_modules/firebase-admin/lib/utils/error.js:149:16)
    at /var/task/node_modules/firebase-admin/lib/auth/token-verifier.js:177:39
    at processTicksAndRejections (internal/process/task_queues.js:93:5) {
  errorInfo: {
    code: 'auth/argument-error',
    message: 'Firebase ID token has "kid" claim which does not correspond to a known public key. Most likely the ID token is expired, so get a fresh token from your client app and try again.'
  },
  codePrefix: 'auth'
}
2021-03-16T04:29:45.461Z	a5539b48-1846-4b8a-8914-d28700cb65a6	ERROR	FirebaseAuthError: Firebase ID token has "kid" claim which does not correspond to a known public key. Most likely the ID token is expired, so get a fresh token from your client app and try again.
    at FirebaseAuthError.FirebaseError [as constructor] (/var/task/node_modules/firebase-admin/lib/utils/error.js:44:28)
    at FirebaseAuthError.PrefixedFirebaseError [as constructor] (/var/task/node_modules/firebase-admin/lib/utils/error.js:90:28)
    at new FirebaseAuthError (/var/task/node_modules/firebase-admin/lib/utils/error.js:149:16)
    at /var/task/node_modules/firebase-admin/lib/auth/token-verifier.js:177:39
    at processTicksAndRejections (internal/process/task_queues.js:93:5) {
  errorInfo: {
    code: 'auth/argument-error',
    message: 'Firebase ID token has "kid" claim which does not correspond to a known public key. Most likely the ID token is expired, so get a fresh token from your client app and try again.'
  },
  codePrefix: 'auth'
}
2021-03-16T04:29:45.462Z	a5539b48-1846-4b8a-8914-d28700cb65a6	ERROR	Unhandled Promise Rejection 	{"errorType":"Runtime.UnhandledPromiseRejection","errorMessage":"Error: Firebase ID token has \"kid\" claim which does not correspond to a known public key. Most likely the ID token is expired, so get a fresh token from your client app and try again.","reason":{"code":"auth/argument-error","message":"Firebase ID token has \"kid\" claim which does not correspond to a known public key. Most likely the ID token is expired, so get a fresh token from your client app and try again."},"promise":{},"stack":["Runtime.UnhandledPromiseRejection: Error: Firebase ID token has \"kid\" claim which does not correspond to a known public key. Most likely the ID token is expired, so get a fresh token from your client app and try again.","    at process.<anonymous> (/var/runtime/index.js:35:15)","    at process.emit (events.js:327:22)","    at processPromiseRejections (internal/process/promises.js:245:33)","    at processTicksAndRejections (internal/process/task_queues.js:94:32)"]}
Unknown application error occurred
```

**Version**
```json
"next-firebase-auth": "^0.13.0-alpha.0",
```

**To Reproduce**
I haven't been able to reproduce this bug on my end, so not sure what's causing it.

**Expected behavior**
No 500 error 😃

**Additional context**

Here are the exports for the SSR page that is causing the error:
```tsx
const getServerSideProps = withAuthUserTokenSSR({
  whenAuthed: AuthAction.RENDER,
  whenUnauthed: AuthAction.RENDER,
})(async ({ AuthUser }) => {
  const topSelling = await getTopSelling({
    query: `available_for_sale:true`,
  });
  const carouselSlides = await getAllSlides();

  return {
    props: {
      topSelling,
      carouselSlides,
    },
  };
});

export { getServerSideProps };
export default withAuthUser({
  whenAuthed: AuthAction.RENDER,
  whenUnauthedBeforeInit: AuthAction.RENDER,
  whenUnauthedAfterInit: AuthAction.RENDER,
})(CartPage);
```

I'm also using `useAuthUser` hook in a nested component, but I don't _think_ it's related to this error.

Photo of error page from user:
![20210316_153010](https://user-images.githubusercontent.com/3422401/111257365-1b278a00-866f-11eb-91e5-fd3653ce301c.jpg)

**Comment by kmjennison** [comment]
I'm not sure if this is related to this package or not—tough to tell without a reproduction. This package should be automatically refreshing the ID token. Can you confirm you're getting an ID token on every API call and not, e.g., storing the token in state or cache? If the latter, this error would appear after the app's been open for an hour (when the ID token expires).

From [Stack Overflow](https://stackoverflow.com/a/47805455):
> Also just in case, ensure your server clock is synchronized. It is unlikely but your clock could be out of sync for some reason.

**Comment by kmjennison** [comment]
Closing until there's a way to reproduce this, but feel free to comment, and I'll help where possible.

**Comment by lukebennett88** [comment]
That's for getting back to me Kevin.
User accounts weren't strictly required for this site, so for the time being I've removed them completely.
When I try to add the back, I'll do some more testing and report back.

**Comment by sdeboeck** [comment]
I came across the same error, my workaround was to clear my cookies and a new refresh token and id token would be created. I haven't been able to reproduce it afterwards, so that's a bummer :face_with_head_bandage: 

Going trough the flow of withAuthUserTokenSSR, the only place where AuthUserTokens are used, I bumped on [verifyIdToken](https://github.com/gladly-team/next-firebase-auth/blob/63563ad2913c402802bacb204cb9920d9df260ed/src/firebaseAdmin.js#L48). An error is thrown if it's something else then auth/id-token-expired. But in this example the kid was expired, resulting in auth/argument-error.
